pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py

```diff
@@ -243,55 +209,47 @@ def configure_optimizers(self):
-        parser.add_argument('--batch_size', type=int, default=64)
-        parser.add_argument('--pretrained', type=str, default=None)
-        parser.add_argument('--data_dir', type=str, default=os.getcwd())
-        parser.add_argument('--num_workers', type=int, default=8, help="num dataloader workers")
-
-        parser.add_argument('--learning_rate', type=float, default=1e-3)
+        parser.add_argument(
+            "--hidden_dim",
+            type=int,
+            default=128,
+            help="itermediate layers dimension before embedding for default encoder/decoder",
+        )
+        parser.add_argument("--latent_dim", type=int, default=4, help="dimension of latent variables z")
+        parser.add_argument("--pretrained", type=str, default=None)
+        parser.add_argument("--learning_rate", type=float, default=1e-3)
         return parser
 
 
-def cli_main():
+def cli_main(args=None):
     from pl_bolts.callbacks import LatentDimInterpolator, TensorboardGenerativeModelImageSampler
-    from pl_bolts.datamodules import ImagenetDataModule
 
-    pl.seed_everything(1234)
+    # cli_main()
     parser = ArgumentParser()
-    parser.add_argument('--dataset', default='mnist', type=str, help='mnist, stl10, imagenet')
-
+    parser.add_argument("--dataset", default="mnist", type=str, help="mnist, cifar10, stl10, imagenet")
+    script_args, _ = parser.parse_known_args(args)
+
+    if script_args.dataset == "mnist":
+        dm_cls = MNISTDataModule
+    elif script_args.dataset == "cifar10":
+        dm_cls = CIFAR10DataModule
+    elif script_args.dataset == "stl10":
+        dm_cls = STL10DataModule
+    elif script_args.dataset == "imagenet":
+        dm_cls = ImagenetDataModule
+
+    parser = dm_cls.add_argparse_args(parser)
     parser = pl.Trainer.add_argparse_args(parser)
     parser = VAE.add_model_specific_args(parser)
-    parser = ImagenetDataModule.add_argparse_args(parser)
-    parser = MNISTDataModule.add_argparse_args(parser)
-    args = parser.parse_args()
-
-    # default is mnist
-    datamodule = None
-    if args.dataset == 'imagenet2012':
-        datamodule = ImagenetDataModule.from_argparse_args(args)
-    elif args.dataset == 'stl10':
-        datamodule = STL10DataModule.from_argparse_args(args)
+    args = parser.parse_args(args)
 
+    dm = dm_cls.from_argparse_args(args)
+    model = VAE(*dm.size(), **vars(args))
     callbacks = [TensorboardGenerativeModelImageSampler(), LatentDimInterpolator(interpolate_epoch_interval=5)]
-    vae = VAE(**vars(args), datamodule=datamodule)
-    trainer = pl.Trainer.from_argparse_args(
-        args,
-        callbacks=callbacks,
-        progress_bar_refresh_rate=10,
-    )
-    trainer.fit(vae)
+    trainer = pl.Trainer.from_argparse_args(args, callbacks=callbacks, progress_bar_refresh_rate=20)
+    trainer.fit(model, dm)
+    return dm, model, trainer
 
 
-if __name__ == '__main__':
-    cli_main()
+if __name__ == "__main__":
+    dm, model, trainer = cli_main()
```

Comment:
```suggestion
    cli_main()
```

Reply:
I often run `python -i some_script.py --somearg somevalue` when running CLI scripts so I have access to the variables after its done running if I forgot to save something/need to investigate variables. 

Lets leave it the way it is so we have failsafe when running interactively. I see no cons, only pros for leaving it.